nautobot/circuits/migrations/0004_providernetworks.py

```diff
@@ -0,0 +1,142 @@
+                related_name="circuit_terminations",
+                to="dcim.site",
+            ),
+        ),
+        migrations.CreateModel(
+            name="ProviderNetwork",
+            fields=[
+                (
+                    "id",
+                    models.UUIDField(
+                        default=uuid.uuid4, editable=False, primary_key=True, serialize=False, unique=True
+                    ),
+                ),
+                ("created", models.DateField(auto_now_add=True, null=True)),
+                ("last_updated", models.DateTimeField(auto_now=True, null=True)),
+                (
+                    "_custom_field_data",
+                    models.JSONField(blank=True, default=dict, encoder=django.core.serializers.json.DjangoJSONEncoder),
+                ),
+                ("name", models.CharField(max_length=100)),
+                ("description", models.CharField(blank=True, max_length=200)),
+                ("comments", models.TextField(blank=True)),
+                (
+                    "provider",
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.PROTECT,
+                        related_name="provider_networks",
+                        to="circuits.provider",
+                    ),
+                ),
+                (
+                    "tags",
+                    taggit.managers.TaggableManager(
+                        help_text="A comma-separated list of tags.",
+                        through="extras.TaggedItem",
+                        to="extras.Tag",
+                        verbose_name="Tags",
+                    ),
+                ),
+            ],
+            options={
+                "ordering": ("provider", "name"),
+            },
+        ),
+        migrations.AddField(
+            model_name="circuittermination",
+            name="provider_network",
+            field=models.ForeignKey(
+                blank=True,
+                null=True,
+                on_delete=django.db.models.deletion.PROTECT,
+                related_name="circuit_terminations",
+                to="circuits.providernetwork",
+            ),
+        ),
+        migrations.AddConstraint(
+            model_name="providernetwork",
+            constraint=models.UniqueConstraint(
+                fields=("provider", "name"), name="circuits_providernetwork_provider_name"
+            ),
+        ),
+        migrations.AlterUniqueTogether(
+            name="providernetwork",
+            unique_together={("provider", "name")},
+        ),
+        migrations.RunPython(
+            code=cache_circuit_terminations,
+            reverse_code=django.db.migrations.operations.special.RunPython.noop,
+        ),
+        migrations.AddField(
+            model_name="providernetwork",
+            name="slug",
+            field=nautobot.core.fields.AutoSlugField(blank=True, max_length=100, populate_from="name", unique=True),
+        ),
+    ]
```

Comment:
It's not immediately clear to me why the `AddField` for `provider network.slug` is after the `RunPython` data migration. Was this intentional or can we get away with leaving this in this migration file and moving the `RunPython` to its own migration file that immediately follows these schema changes?

I'm asking because in order to split this it would have to be either:

1. Add `ProviderNetwork` in `0004_providernetwork`
2. Run `cache_circuit_terminations` data migration in `0005_cache_circuit_terminations`
3. Run `AddField` to add `ProviderNetwork.slug` in `0006_providernetwork_slug`

Or ideally, just the first two:

1. Add `ProviderNetwork` (including `slug`) in `0004_providernetwork`
2. Run `cache_circuit_terminations` data migration in `0005_cache_circuit_terminations`

Reply:
This was not intentional, I fixed it as you mentioned.